src/main/webapp/WEB-INF/web.xml

```diff
@@ -37,6 +37,19 @@
         <param-name>primefaces.FONT_AWESOME</param-name>
         <param-value>true</param-value>
     </context-param>
+    
+    <context-param>
+        <param-name>primefaces.CSP</param-name>
+        <param-value>true</param-value>
+    </context-param>
+
+    <session-config>
+        <cookie-config>
+            <http-only>true</http-only>
+            <!--Breaks PrimeFaces: <secure>true</secure>-->
```

Comment:
or because SSL termination is not done by the server.

Reply:
@rdebusscher True. I have Nginx there on my box doing a proxy.